src/lib/utils/SynapseContext.tsx

```diff
@@ -33,19 +39,20 @@ export type SynapseContextProviderProps = {
  * @param param0
  * @returns
  */
-export const SynapseContextProvider: React.FunctionComponent<SynapseContextProviderProps> = ({
-  children,
-  synapseContext,
-  queryClient,
-}) => {
-  return (
-    <SynapseContext.Provider value={synapseContext}>
-      <QueryClientProvider client={queryClient ?? defaultQueryClient}>
-        {children}
-      </QueryClientProvider>
-    </SynapseContext.Provider>
-  )
-}
+export const SynapseContextProvider: React.FunctionComponent<SynapseContextProviderProps> =
+  ({ children, synapseContext, queryClient }) => {
+    return (
+      <SynapseContext.Provider value={synapseContext}>
+        <QueryClientProvider client={queryClient ?? defaultQueryClient}>
+          {synapseContext?.withErrorBoundary ? (
+            <SynapseErrorBoundary>{children}</SynapseErrorBoundary>
+          ) : (
+            children
+          )}
```

Comment:
I think the main reason we haven't yet gotten much value out of error boundaries is because we treat SRC like a component library, and error boundaries solve problems for complete applications. Placement of error boundaries and the design of fallback components can be done more deliberately in a hierarchical app. Since we don't have many complex hierarchies, it often doesn't make sense to put error boundaries in SRC components, but it may make sense for these components to be surrounded within an error boundary.

As an alternative to adding an error boundary wrapper to every component, I've added the ability to add the error boundary where we provide the SynapseContext. In SWC, we wrap nearly every component in individual `SynapseContextProvider`s, so it seems valuable to place an error boundary here.

The error boundary is optional and not added by default because this makes no sense in React apps like Portals, where we want this context provider to be placed as high up in the component tree as possible. In that case, it makes no sense to wrap an entire app in an error boundary, since the whole app would visibly break any time an uncaught error is thrown.